theme/templates/resource-landing-page/modals.html

```diff
@@ -1393,6 +1397,60 @@ <h4 class="modal-title" id="add-relation"> Add Relation </h4>
         </div>
     </div>
 
+    <div class="modal fade" id="customize-citation-dialog" tabindex="-1" role="dialog" aria-labelledby="customize-citation"
+         aria-hidden="true">
+        <div class="modal-dialog">
+            <div class="modal-content">
+                {% if custom_citation %}
+                    <form action="/hsapi/_internal/{{ cm.short_id }}/citation/{{ citation_id }}/update-metadata/" method="POST"
+                          enctype="multipart/form-data">
+                {% else %}
+                    <form action="/hsapi/_internal/{{ cm.short_id }}/citation/add-metadata/" method="POST"
+                          enctype="multipart/form-data">
+                {% endif %}
+                <input type="hidden" name="resource-mode">
+                    {% csrf_token %}
+                    <div class="modal-header">
+                        <button type="button" class="close" data-dismiss="modal" aria-hidden="true">&times;</button>
+                        <h4 class="modal-title" id="customize-citation">Custom Citation</h4>
+                    </div>
+                    <div class="modal-body">
+                        <div class="form-group">
+                            <label for="custom-citation-input">Provide the full text of how you would like this resource to be cited:</label>
+                            <textarea class="form-control" name="content" id="custom-citation-input" cols="30"
+                                      rows="3">{% if custom_citation %}{{ custom_citation }}{% endif %}</textarea>
+                            Suggested Format: Creator(s) (Publication Year). Title,  Version, Publisher, Resource Type, Identifier (preferably encoded as a URL)
+                        </div>
+
+                    </div>
+                    <div class="modal-footer">
+                        {% if custom_citation %}
+                        <script language="JavaScript">
+                          function AsyncRevertCitation() {
+                            document.body.style.cursor = 'wait';
+                            $.ajax({
+                              type: "POST",
+                              url: '/hsapi/_internal/{{cm.short_id}}/citation/{{citation_id}}/delete-metadata/',
+                            }).complete(function() {
+                              document.body.style.cursor = 'default';
+                              if (window.location.href.includes('editing=True')) {
```

Comment:
ah, this is where the editing=True came from.  I don't think we should be introducing this parameter.  It's not done anywhere else.

Reply:
If you can suggest a different way to refresh the Citation information, without a page reload or lots of additional backend/frontend rewrites, I'm open to that approach. Being tied to the Mezzanine Page for the resource data as well as the @page_processor is undesirable. Regardless of what we do here, we should mark this as a discussion to have soon.

Reply:
A POST with resource-mode:edit in the body is the way the "edit" button on the resource landing page does it.  I believe mezzanine also allows you to use that parameter as query parameters in a GET too.

Reply:
This may all change, given the flow through a new `_internal` endpoint, but the reasoning here was that a reload call in pure JavaScript here didn't seem to trigger popup blockers or new tabs, whereas a call from within the JS here to a new POST/GET object might be discouraged or literally throw a popup blocker/new tab. I had looked into manually rewriting the DOM, based on how Django responds with a template, since this isn't a RESTful endpoint... I'm pretty sure I tested that, though. Can visit again.

Reply:
This should probably stay as-is for now.

Reply:
I do not see a reason to leave this as is.  We already have the ability to enter edit mode with a GET request.

Reply:
There's two places where this is required and while one might work the other definitely will not work across browsers, without a risky rewrite. The existing code isn't designed to do what we're asking it, so things like this are necessary.

Reply:
I had been putting your recommended text in the POST, you have clarified to put it in the async response GET. Updated.

Reply:
The ajax POST on line 1433? An ajax call cannot redirect and is for asynchronous calls.

Reply:
The proper way to post with async JavaScript is to supply all the information upfront, including special instructions, such as a querystring about page origin or destination after the post. The response is handled in the complete section. Typically, response data contains only relevant information based on POST actions. In the case of Django templates, the response contains the entire rendered page, which can be interpreted in the complete but I already explained to you the direct manipulation of the html page with that response was risky cross-browser and had opted for a refresh. The hack to send a second GET with the edit querystring is a nice hack, and happens to work the same as the refresh. Hopefully that is all clear now.

Reply:
This allowed us to delete three lines of code and hopefully wrap up this review. Jeff has requested some styling changes, so I will be coding these today.